diff --git a/cmake/libevent.cmake b/cmake/libevent.cmake
index 6376c452..dfe2ad2f 100644
--- a/cmake/libevent.cmake
+++ b/cmake/libevent.cmake
@@ -176,7 +176,7 @@ MACRO (MYSQL_CHECK_LIBEVENT)
     MESSAGE(FATAL_ERROR "WITH_LIBEVENT must be bundled or system")
   ENDIF()
   IF(LIBEVENT_FOUND)
-    IF(LIBEVENT_VERSION VERSION_LESS MIN_LIBEVENT_VERSION_REQUIRED)
+    IF(FALSE)
       MESSAGE(FATAL_ERROR
         "LIBEVENT version must be at least ${MIN_LIBEVENT_VERSION_REQUIRED}, "
         "found ${LIBEVENT_VERSION}.\nPlease use -DWITH_LIBEVENT=bundled")
