diff --git a/ports/libmysql/portfile.cmake b/ports/libmysql/portfile.cmake
index 3a08e3a25..bd6fef3bf 100644
--- a/ports/libmysql/portfile.cmake
+++ b/ports/libmysql/portfile.cmake
@@ -19,6 +19,7 @@ vcpkg_from_github(
         Add-target-include-directories.patch
         homebrew.patch
         fix_dup_symbols.patch
+        mysql_patch.patch
 )

 file(REMOVE_RECURSE "${SOURCE_PATH}/include/boost_1_70_0")
@@ -70,39 +71,6 @@ vcpkg_cmake_configure(

 vcpkg_cmake_install(ADD_BIN_TO_PATH)

-list(APPEND MYSQL_TOOLS
-    comp_err
-    my_print_defaults
-    mysql
-    mysql_config_editor
-    mysql_migrate_keyring
-    mysql_secure_installation
-    mysql_ssl_rsa_setup
-    mysqladmin
-    mysqlbinlog
-    mysqlcheck
-    mysqldump
-    mysqlimport
-    mysqlpump
-    mysqlshow
-    mysqlslap
-    mysqltest
-    perror
-    zlib_decompress
-)
-
-if (VCPKG_TARGET_IS_WINDOWS)
-    list(APPEND MYSQL_TOOLS
-        echo
-    )
-else()
-    list(APPEND MYSQL_TOOLS
-        mysql_config
-    )
-endif()
-
-vcpkg_copy_tools(TOOL_NAMES ${MYSQL_TOOLS} AUTO_CLEAN)
-
 file(RENAME "${CURRENT_PACKAGES_DIR}/share" "${CURRENT_PACKAGES_DIR}/${PORT}")
 file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/share")
 file(RENAME "${CURRENT_PACKAGES_DIR}/${PORT}" "${CURRENT_PACKAGES_DIR}/share/${PORT}")
