diff --git a/cmake/build_configurations/compiler_options.cmake b/cmake/build_configurations/compiler_options.cmake
index 5dd2499c..f60d2b67 100644
--- a/cmake/build_configurations/compiler_options.cmake
+++ b/cmake/build_configurations/compiler_options.cmake
@@ -64,12 +64,6 @@ IF(UNIX)
     SET(COMMON_CXX_FLAGS             "-std=c++17 -fno-omit-frame-pointer")
   ENDIF()
 
-  # Faster TLS model
-  IF(MY_COMPILER_IS_GNU_OR_CLANG AND NOT SOLARIS AND NOT LINUX_RHEL6)
-    STRING_APPEND(COMMON_C_FLAGS     " -ftls-model=initial-exec")
-    STRING_APPEND(COMMON_CXX_FLAGS   " -ftls-model=initial-exec")
-  ENDIF()
-
   # Use STRING_PREPEND here, so command-line input can override our defaults.
   STRING_PREPEND(CMAKE_C_FLAGS                  "${COMMON_C_FLAGS} ")
   STRING_PREPEND(CMAKE_C_FLAGS_RELWITHDEBINFO   "${SECTIONS_FLAG} ")
diff --git a/configure.cmake b/configure.cmake
index 96664b16..99e26446 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -403,21 +403,8 @@ MY_CHECK_CXX_COMPILER_FLAG("-fvisibility=hidden" HAVE_VISIBILITY_HIDDEN)
 # Code tests
 #
 
-CHECK_C_SOURCE_RUNS("
-#include <time.h>
-int main()
-{
-  struct timespec ts;
-  return clock_gettime(CLOCK_MONOTONIC, &ts);
-}" HAVE_CLOCK_GETTIME)
-
-CHECK_C_SOURCE_RUNS("
-#include <time.h>
-int main()
-{
-  struct timespec ts;
-  return clock_gettime(CLOCK_REALTIME, &ts);
-}" HAVE_CLOCK_REALTIME)
+set(HAVE_CLOCK_GETTIME 1)
+set(HAVE_CLOCK_REALTIME 1)
 
 IF(NOT STACK_DIRECTION)
   IF(CMAKE_CROSSCOMPILING)
diff --git a/libmysql/CMakeLists.txt b/libmysql/CMakeLists.txt
index bc900c58..475559a2 100644
--- a/libmysql/CMakeLists.txt
+++ b/libmysql/CMakeLists.txt
@@ -255,10 +255,10 @@ ELSE()
   FIND_LIBRARY(RESOLV_LIBRARY NAMES resolv)
   IF (RESOLV_LIBRARY)
     LIST(APPEND LIBS_TO_LINK ${RESOLV_LIBRARY})
-    SET(HAVE_UNIX_DNS_SRV 1 PARENT_SCOPE)
-    SET(HAVE_DNS_SRV 1)
-    MESSAGE(STATUS "Found Unix DNS SRV APIs")
   ENDIF()
+  SET(HAVE_UNIX_DNS_SRV 1 PARENT_SCOPE)
+  SET(HAVE_DNS_SRV 1)
+  MESSAGE(STATUS "Found Unix DNS SRV APIs")
 ENDIF()
 
 IF(HAVE_DNS_SRV EQUAL 0)
