diff --git a/cmake/floating_point.cmake b/cmake/floating_point.cmake
index c460e3d9..21d18ee5 100644
--- a/cmake/floating_point.cmake
+++ b/cmake/floating_point.cmake
@@ -51,9 +51,4 @@ SET(code "
 CMAKE_PUSH_CHECK_STATE()
 STRING_APPEND(CMAKE_REQUIRED_FLAGS " -O3")
 
-IF(MY_COMPILER_IS_GNU)
-  CHECK_C_SOURCE_RUNS("${code}" HAVE_C_FLOATING_POINT_FUSED_MADD)
-  CHECK_CXX_SOURCE_RUNS("${code}" HAVE_CXX_FLOATING_POINT_FUSED_MADD)
-ENDIF()
-
 CMAKE_POP_CHECK_STATE()
diff --git a/configure.cmake b/configure.cmake
index 99e26446..d71f522b 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -311,16 +311,7 @@ CHECK_FUNCTION_EXISTS (timer_settime HAVE_TIMER_SETTIME)
 CHECK_FUNCTION_EXISTS (kqueue HAVE_KQUEUE)
 
 # Check whether the setns() API function supported by a target platform
-CHECK_C_SOURCE_RUNS("
-#ifndef _GNU_SOURCE
-#define _GNU_SOURCE
-#endif
-#include <sched.h>
-int main()
-{
-  (void)setns(0, 0);
-  return 0;
-}" HAVE_SETNS)
+SET(HAVE_SETNS 0)
 
 CHECK_SYMBOL_EXISTS(EVFILT_TIMER "sys/types.h;sys/event.h;sys/time.h" HAVE_EVFILT_TIMER)
 IF(HAVE_KQUEUE AND HAVE_EVFILT_TIMER)
diff --git a/libmysql/CMakeLists.txt b/libmysql/CMakeLists.txt
index 475559a2..77735719 100644
--- a/libmysql/CMakeLists.txt
+++ b/libmysql/CMakeLists.txt
@@ -255,6 +255,8 @@ ELSE()
   FIND_LIBRARY(RESOLV_LIBRARY NAMES resolv)
   IF (RESOLV_LIBRARY)
     LIST(APPEND LIBS_TO_LINK ${RESOLV_LIBRARY})
+  ELSE()
+    LIST(APPEND LIBS_TO_LINK resolv)    
   ENDIF()
   SET(HAVE_UNIX_DNS_SRV 1 PARENT_SCOPE)
   SET(HAVE_DNS_SRV 1)
diff --git a/scripts/CMakeLists.txt b/scripts/CMakeLists.txt
index 987870e9..b0d31db6 100644
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -24,7 +24,7 @@ DISABLE_MISSING_PROFILE_WARNING()
 
 ADD_SUBDIRECTORY(sys_schema)
 
-MYSQL_ADD_EXECUTABLE(comp_sql comp_sql.cc SKIP_INSTALL)
+MYSQL_ADD_EXECUTABLE(comp_sql_unused comp_sql.cc SKIP_INSTALL)
 
 # Add compiler information, for easier debugging of output from mysql_config.
 SET(COMPILER_ID_AND_VERSION
@@ -62,7 +62,7 @@ ADD_CUSTOM_COMMAND(
   mysql_fix_privilege_tables.sql
   mysql_fix_privilege_tables_sql.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables.sql
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_fix.sql
 )
@@ -80,7 +80,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_sys_schema.sql
   sql_commands_sys_schema.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_sys_schema.sql
   )
 
@@ -91,7 +91,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_data.sql
   sql_commands_system_data.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_data.sql
 )
 
@@ -102,7 +102,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_data_fix.sql
   sql_commands_system_tables_data_fix.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_data_fix.sql
 )
 
@@ -113,7 +113,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables.sql
   sql_commands_system_tables.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables.sql
 )
 
@@ -124,7 +124,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_users.sql
   sql_commands_system_users.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_users.sql
 )
 
@@ -135,7 +135,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/fill_help_tables.sql
   sql_commands_help_data.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/fill_help_tables.sql
 )
 
@@ -146,7 +146,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/firewall_stored_procedures.sql
   sql_firewall_stored_procedures.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/firewall_stored_procedures.sql
 )
 
diff --git a/scripts/sys_schema/CMakeLists.txt b/scripts/sys_schema/CMakeLists.txt
index d8076220..a033f3ea 100644
--- a/scripts/sys_schema/CMakeLists.txt
+++ b/scripts/sys_schema/CMakeLists.txt
@@ -212,7 +212,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_BINARY_DIR}/ALL_SYS_SCHEMA.sql
   sql_commands.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql all_sys_schema
+  DEPENDS comp_sql_unused all_sys_schema
   ${CMAKE_CURRENT_BINARY_DIR}/ALL_SYS_SCHEMA.sql
   VERBATIM
 )
@@ -222,7 +222,7 @@ MY_ADD_CUSTOM_TARGET(sql_commands
 
 MY_ADD_CUSTOM_TARGET(GenSysSchemaC
   ALL
-  DEPENDS comp_sql sql_commands
+  DEPENDS comp_sql_unused sql_commands
   COMMENT "Generating Sys Schema C code"
 )
 
diff --git a/share/CMakeLists.txt b/share/CMakeLists.txt
index 41f9e8dd..e74e1e49 100644
--- a/share/CMakeLists.txt
+++ b/share/CMakeLists.txt
@@ -47,19 +47,6 @@ SET(dirs
   ukrainian
   )
 
-FOREACH (dir ${dirs})
-  INSTALL(DIRECTORY ${CMAKE_BINARY_DIR}/${INSTALL_MYSQLSHAREDIR}/${dir}
-    DESTINATION ${INSTALL_MYSQLSHAREDIR}
-    COMPONENT Server
-    )
-ENDFOREACH()
-
-INSTALL(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/charsets
-  DESTINATION ${INSTALL_MYSQLSHAREDIR}
-  COMPONENT Server
-  PATTERN "languages.html" EXCLUDE
-)
-
 # Don't copy if we're building in-source.
 IF (NOT ${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
   CONFIGURE_FILE(dictionary.txt
diff --git a/strings/CMakeLists.txt b/strings/CMakeLists.txt
index 2feec0c9..a18ab66e 100644
--- a/strings/CMakeLists.txt
+++ b/strings/CMakeLists.txt
@@ -58,7 +58,7 @@ SET(STRINGS_SOURCES
   xml.cc
 )
 
-MYSQL_ADD_EXECUTABLE(uca9dump uca9-dump.cc SKIP_INSTALL)
+MYSQL_ADD_EXECUTABLE(uca9dump_unused uca9-dump.cc SKIP_INSTALL)
 
 MY_CHECK_CXX_COMPILER_WARNING("-Wmissing-profile" HAS_MISSING_PROFILE)
 IF(HAS_MISSING_PROFILE)
@@ -77,7 +77,7 @@ ADD_CUSTOM_COMMAND(OUTPUT ${ZH_HANS_DST_FILE}
                    COMMAND uca9dump ja
                      --in_file=${JA_HANS_SRC_FILE}
                      --out_file=${JA_HANS_DST_FILE}
-                   DEPENDS uca9dump ${ZH_HANS_SRC_FILE} ${JA_HANS_SRC_FILE}
+                   DEPENDS uca9dump_unused ${ZH_HANS_SRC_FILE} ${JA_HANS_SRC_FILE}
                   )
 
 SET_SOURCE_FILES_PROPERTIES(
diff --git a/utilities/CMakeLists.txt b/utilities/CMakeLists.txt
index c0dfac56..95e7c417 100644
--- a/utilities/CMakeLists.txt
+++ b/utilities/CMakeLists.txt
@@ -24,14 +24,14 @@ ADD_WSHADOW_WARNING()
 DISABLE_MISSING_PROFILE_WARNING()
 
 IF(NOT CMAKE_CROSSCOMPILING)
-  MYSQL_ADD_EXECUTABLE(comp_err
+  MYSQL_ADD_EXECUTABLE(comp_err_unused
     comp_err.cc
     COMPONENT Test
     LINK_LIBRARIES mysys
     )
 ENDIF()
 
-MYSQL_ADD_EXECUTABLE(comp_client_err
+MYSQL_ADD_EXECUTABLE(comp_client_err_unused
   comp_client_err.cc
   ${CMAKE_SOURCE_DIR}/libmysql/errmsg.cc
   LINK_LIBRARIES mysys
@@ -102,8 +102,8 @@ ADD_DEFINITIONS(-DMUTEX_EVENT)
 # The copied shared libraries must be available at build time.
 IF(TARGET copy_openssl_dlls)
   ADD_DEPENDENCIES(GenError copy_openssl_dlls)
-  ADD_DEPENDENCIES(comp_client_err copy_openssl_dlls)
-  ADD_DEPENDENCIES(comp_err copy_openssl_dlls)
+  ADD_DEPENDENCIES(comp_client_err_unused copy_openssl_dlls)
+  ADD_DEPENDENCIES(comp_err_unused copy_openssl_dlls)
 ENDIF()
 
 MYSQL_ADD_EXECUTABLE(static_thread_local_test
