diff --git a/libmysql/CMakeLists.txt b/libmysql/CMakeLists.txt
index 475559a2..77735719 100644
--- a/libmysql/CMakeLists.txt
+++ b/libmysql/CMakeLists.txt
@@ -255,6 +255,8 @@ ELSE()
   FIND_LIBRARY(RESOLV_LIBRARY NAMES resolv)
   IF (RESOLV_LIBRARY)
     LIST(APPEND LIBS_TO_LINK ${RESOLV_LIBRARY})
+  ELSE()
+    LIST(APPEND LIBS_TO_LINK resolv)    
   ENDIF()
   SET(HAVE_UNIX_DNS_SRV 1 PARENT_SCOPE)
   SET(HAVE_DNS_SRV 1)
diff --git a/scripts/CMakeLists.txt b/scripts/CMakeLists.txt
index 987870e9..b0d31db6 100644
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -24,7 +24,7 @@ DISABLE_MISSING_PROFILE_WARNING()
 
 ADD_SUBDIRECTORY(sys_schema)
 
-MYSQL_ADD_EXECUTABLE(comp_sql comp_sql.cc SKIP_INSTALL)
+MYSQL_ADD_EXECUTABLE(comp_sql_unused comp_sql.cc SKIP_INSTALL)
 
 # Add compiler information, for easier debugging of output from mysql_config.
 SET(COMPILER_ID_AND_VERSION
@@ -62,7 +62,7 @@ ADD_CUSTOM_COMMAND(
   mysql_fix_privilege_tables.sql
   mysql_fix_privilege_tables_sql.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables.sql
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_fix.sql
 )
@@ -80,7 +80,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_sys_schema.sql
   sql_commands_sys_schema.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_sys_schema.sql
   )
 
@@ -91,7 +91,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_data.sql
   sql_commands_system_data.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_data.sql
 )
 
@@ -102,7 +102,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_data_fix.sql
   sql_commands_system_tables_data_fix.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables_data_fix.sql
 )
 
@@ -113,7 +113,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables.sql
   sql_commands_system_tables.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_tables.sql
 )
 
@@ -124,7 +124,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_users.sql
   sql_commands_system_users.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/mysql_system_users.sql
 )
 
@@ -135,7 +135,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/fill_help_tables.sql
   sql_commands_help_data.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/fill_help_tables.sql
 )
 
@@ -146,7 +146,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_SOURCE_DIR}/firewall_stored_procedures.sql
   sql_firewall_stored_procedures.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql
+  DEPENDS comp_sql_unused
   ${CMAKE_CURRENT_SOURCE_DIR}/firewall_stored_procedures.sql
 )
 
diff --git a/scripts/sys_schema/CMakeLists.txt b/scripts/sys_schema/CMakeLists.txt
index d8076220..a033f3ea 100644
--- a/scripts/sys_schema/CMakeLists.txt
+++ b/scripts/sys_schema/CMakeLists.txt
@@ -212,7 +212,7 @@ ADD_CUSTOM_COMMAND(
   ${CMAKE_CURRENT_BINARY_DIR}/ALL_SYS_SCHEMA.sql
   sql_commands.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS comp_sql all_sys_schema
+  DEPENDS comp_sql_unused all_sys_schema
   ${CMAKE_CURRENT_BINARY_DIR}/ALL_SYS_SCHEMA.sql
   VERBATIM
 )
@@ -222,7 +222,7 @@ MY_ADD_CUSTOM_TARGET(sql_commands
 
 MY_ADD_CUSTOM_TARGET(GenSysSchemaC
   ALL
-  DEPENDS comp_sql sql_commands
+  DEPENDS comp_sql_unused sql_commands
   COMMENT "Generating Sys Schema C code"
 )
 
diff --git a/strings/CMakeLists.txt b/strings/CMakeLists.txt
index 2feec0c9..a18ab66e 100644
--- a/strings/CMakeLists.txt
+++ b/strings/CMakeLists.txt
@@ -58,7 +58,7 @@ SET(STRINGS_SOURCES
   xml.cc
 )
 
-MYSQL_ADD_EXECUTABLE(uca9dump uca9-dump.cc SKIP_INSTALL)
+MYSQL_ADD_EXECUTABLE(uca9dump_unused uca9-dump.cc SKIP_INSTALL)
 
 MY_CHECK_CXX_COMPILER_WARNING("-Wmissing-profile" HAS_MISSING_PROFILE)
 IF(HAS_MISSING_PROFILE)
@@ -77,7 +77,7 @@ ADD_CUSTOM_COMMAND(OUTPUT ${ZH_HANS_DST_FILE}
                    COMMAND uca9dump ja
                      --in_file=${JA_HANS_SRC_FILE}
                      --out_file=${JA_HANS_DST_FILE}
-                   DEPENDS uca9dump ${ZH_HANS_SRC_FILE} ${JA_HANS_SRC_FILE}
+                   DEPENDS uca9dump_unused ${ZH_HANS_SRC_FILE} ${JA_HANS_SRC_FILE}
                   )
 
 SET_SOURCE_FILES_PROPERTIES(
diff --git a/utilities/CMakeLists.txt b/utilities/CMakeLists.txt
index c0dfac56..557a388b 100644
--- a/utilities/CMakeLists.txt
+++ b/utilities/CMakeLists.txt
@@ -23,21 +23,6 @@
 ADD_WSHADOW_WARNING()
 DISABLE_MISSING_PROFILE_WARNING()
 
-IF(NOT CMAKE_CROSSCOMPILING)
-  MYSQL_ADD_EXECUTABLE(comp_err
-    comp_err.cc
-    COMPONENT Test
-    LINK_LIBRARIES mysys
-    )
-ENDIF()
-
-MYSQL_ADD_EXECUTABLE(comp_client_err
-  comp_client_err.cc
-  ${CMAKE_SOURCE_DIR}/libmysql/errmsg.cc
-  LINK_LIBRARIES mysys
-  SKIP_INSTALL
-  )
-
 ADD_CUSTOM_COMMAND(OUTPUT ${PROJECT_BINARY_DIR}/include/mysqlclient_ername.h
   COMMAND comp_client_err
   --in_file=${PROJECT_SOURCE_DIR}/include/errmsg.h
